x-pack/plugins/observability/server/services/slo/resource_installer.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ClusterPutComponentTemplateRequest,
+  IndicesPutIndexTemplateRequest,
+  IngestPutPipelineRequest,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+
+import {
+  SLO_COMPONENT_TEMPLATE_MAPPINGS_NAME,
+  SLO_COMPONENT_TEMPLATE_SETTINGS_NAME,
+  SLO_INDEX_TEMPLATE_NAME,
+  SLO_INGEST_PIPELINE_NAME,
+} from '../../assets/constants';
+import { sloMappingsTemplate } from '../../assets/component_templates/slo_mappings_template';
+import { sloSettingsTemplate } from '../../assets/component_templates/slo_settings_template';
+import { getSLOIndexTemplate } from '../../assets/index_templates/slo_index_templates';
+import { getSLOPipelineTemplate } from '../../assets/ingest_templates/slo_pipeline_template';
+
+export class ResourceInstaller {
+  constructor(private esClient: ElasticsearchClient, private logger: Logger) {}
+
+  public async ensureCommonResourcesInstalled(): Promise<void> {
+    const alreadyInstalled = await this.areResourcesAlreadyInstalled();
+
+    if (alreadyInstalled) {
+      this.logger.debug(
+        `Skipping installation of resources shared for SLO since they already exist`
+      );
+      return;
+    }
+
+    try {
+      await Promise.all([
+        this.createOrUpdateComponentTemplate({
+          name: SLO_COMPONENT_TEMPLATE_MAPPINGS_NAME,
+          ...sloMappingsTemplate,
+        }),
+
+        this.createOrUpdateComponentTemplate({
+          name: SLO_COMPONENT_TEMPLATE_SETTINGS_NAME,
+          ...sloSettingsTemplate,
+        }),
+      ]);
+
+      await this.createOrUpdateIndexTemplate({
+        name: SLO_INDEX_TEMPLATE_NAME,
+        ...getSLOIndexTemplate(`${SLO_INDEX_TEMPLATE_NAME}-*`, [
+          SLO_COMPONENT_TEMPLATE_MAPPINGS_NAME,
+          SLO_COMPONENT_TEMPLATE_SETTINGS_NAME,
+        ]),
+      });
+
+      await this.createOrUpdateIngestPipelineTemplate({
+        id: SLO_INGEST_PIPELINE_NAME,
+        ...getSLOPipelineTemplate(`${SLO_INDEX_TEMPLATE_NAME}-default-`),
```

Comment:
❓ Do we want to override `default` dynamically? If so, where should I take the value from?

Reply:
That should match the current Kibana Namespace. I would need to poke around a bit to find out how to set that.

Reply:
We would also append `-version-1-` so that it matches the version in the mappings.

Reply:
> That should match the current Kibana Namespace. I would need to poke around a bit to find out how to set that.
Found a way